src/cmds/mem/infomem.c

```diff
@@ -0,0 +1,129 @@
+		    		pte_allocator->free / pte_allocator->page_size);
+		} else {
+		    printf("PTE do not available in this configuration\n");
+		}
+	}
+	printf("-----------------------------------------\n");
+}
+
+static void print_phymem(struct page_allocator *allocator) {
+	char *const phymem_alloc_start = phymem_allocated_start();
+	char *const phymem_alloc_end = phymem_allocated_end();
+	const size_t mem_len = phymem_alloc_end - phymem_alloc_start;
+
+	printf("\nPHYSICAL MEMORY:\n");
+	printf("start=%p end=%p size=%zu\n",
+			phymem_alloc_start, phymem_alloc_end, mem_len);
+	printf("-----------------pages-------------------\n");
+	printf("first usable phy_page adr - %p\n", allocator->pages_start);
+	printf("pages count / free - %zu / %zu\n", allocator->pages_n,
+				allocator->free / allocator->page_size);
+	printf("-----------------------------------------\n");
+}
+
+static void print_sections(void) {
+	extern char _text_vma, _rodata_vma, _data_vma, _bss_vma;
+	extern char _text_len, _rodata_len, _data_len, _bss_len, _bss_len_with_reserve;
+	extern char _bss_end, _reserve_end;
+
+	printf("\nSECTIONS:\n");
+	printf("TEXT:   begin - %p, end - %p, len - %p\n",
+			&_text_vma, (int) &_text_vma + &_text_len, &_text_len);
+	printf("RODATA: begin - %p, end - %p, len - %p\n",
+			&_rodata_vma, (int) &_rodata_vma + &_rodata_len, &_rodata_len);
+	printf("DATA:   begin - %p, end - %p, len - %p\n",
+			&_data_vma, (int) &_data_vma + &_data_len, &_data_len);
+	printf("BSS:    begin - %p, end - %p, len - %p\n        "
+			"reserve - %p, end_with_reserve - %p, len_with_reserve - %p\n",
+			&_bss_vma, &_bss_end, &_bss_len,
+			-(int) &_bss_end + &_reserve_end,
+			&_reserve_end, &_bss_len_with_reserve);
+
+	printf("-----------------------------------------\n\n");
+}
+
+static void print_segments(void) {
+	int size = periph_mem_size();
+
+	printf("\nPERIPH SEGMENTS:\n");
+
+	for (int i = 0; i < size; i++) {
+		struct periph_memory_desc desk;
+		periph_elem_desc(&desk, i);
+		printf("seg_num %d: start - %x, end - %x, len - %x\n",
+				i, desk.start, desk.start + desk.len, desk.len);
+	}
+
+	printf("-----------------------------------------\n");
+}
+
+static void print_usage(void) {
+	printf("Usage: infomem [-hpsvi]\n");
+}
+
+static void show_all(void) {
+	print_phymem(phy_allocator());
+	print_vmem(get_pgd_allocator(), get_pmd_allocator(), get_pte_allocator());
+	print_segments();
+	print_sections();
+}
+
+int main(int argc, char **argv) {
+	int opt;
+
+	while (-1 != (opt = getopt(argc, argv, "h:p:s:v:i"))) {
+		switch (opt) {
+		case 'h':
+			print_usage();
+			return 0;
+		case 'p':
+			print_phymem(phy_allocator());
+			return 0;
+		case 's':
+			print_sections();
+			return 0;
+		case 'v':
+			print_vmem(get_pgd_allocator(), get_pmd_allocator(), get_pte_allocator());
+			return 0;
+		case 'i':
+			print_segments();
+			return 0;
+		default:
+			printf("unknown argument!\n");
+			return 0;
```

Comment:
Since it's an unknown parameter, you should return `-1` to indicate failure. https://www.gnu.org/software/coreutils/manual/coreutils.html#Exit-status

Reply:
Fixed.